Sat Aug 26 14:38:22 CEST 2000  Jan Hubicka  <jh@suse.cz>

	* combine.c (make_extraction): Fix rtx_cost comparison to
	match the comment.

*** gcc/combine.c.jj	Sat Aug 26 11:41:35 2000
--- gcc/combine.c	Sat Aug 26 14:36:38 2000
*************** make_extraction (mode, inner, pos, pos_r
*** 6000,6008 ****
  	  rtx temp = gen_rtx_ZERO_EXTEND (mode, new);
  	  rtx temp1 = gen_rtx_SIGN_EXTEND (mode, new);
  
! 	  /* Prefer ZERO_EXTENSION, since it gives more information to
  	     backends.  */
! 	  if (rtx_cost (temp, SET) < rtx_cost (temp1, SET))
  	    return temp;
  	  return temp1;
  	}
--- 6000,6008 ----
  	  rtx temp = gen_rtx_ZERO_EXTEND (mode, new);
  	  rtx temp1 = gen_rtx_SIGN_EXTEND (mode, new);
  
! 	  /* Prefer ZERO_EXTENSION, since it gives more information to
  	     backends.  */
! 	  if (rtx_cost (temp, SET) <= rtx_cost (temp1, SET))
  	    return temp;
  	  return temp1;
  	}

